vendor_data.py

```diff
@@ -56,101 +52,190 @@ def __init__(self, VendorName, TargLang, WordRate, VendorMail = "", CatTool = "X
             CatTool should be a string, if not a TypeError is raised.
             CatTool should be one of the options in the list "CatTools", if not a ValueError is raised.
         """
+        regex_mail = "^[a-z0-9]+[\._]?[a-z0-9]+[@]\w+[.]\w{2,3}$" #regex used to validate email
+        
         
+        if type(ProjectName) != str:
+            raise TypeError("ProjectName should be a string!")
+        else:
+            self.ProjectName = ProjectName
+        if type(SourceLang) != str:
+            raise TypeError("SourceLang should be a string!")
+        else:
+            self.SourceLang = SourceLang
         if type(VendorName) !=str:
             raise TypeError("VendorName should be a string!")
+        else:
+            self.VendorName = VendorName
+            
         if type(TargLang) != str:
             raise TypeError("TargLang should be a string!")
-        if type(WordRate) != float:
-            raise TypeError("WordRate should be a float!")
+        else:
+            self.TargLang = TargLang
+            
         if WordRate:
+            if type(WordRate) != float:
+                raise TypeError("WordRate should be a float!")
             if WordRate > 0.15:
-                raise ValueError("This vendor is too expensive, consider another one.")
-        elif WordRate == 0.00:
-            raise ValueError("Word rate cannot be 0.00.")
+                raise ValueError("This vendor is too expensive, pick another one.")
+            if WordRate == 0.00:
+                raise ValueError("Word rate cannot be 0.00.")
         else:

```

Comment:
This `else` is aligned with `if WordRate`, meaning that `self.WordRate = WordRate` only runs if you _don't_ have `WordRate`.

You should actually remove the `else` and leave `self.WordRate = WordRate` aligned with the `if`s in lines 77-81.
That way, the logic goes like so:
- if there is a value for `WordRate`:
    - if it is not a float, throw an error
    - (otherwise) if it is larger than 0.15, throw an error
    - (otherwise) if it is equal to 0, throw an error
    - (otherwise) assign its value to `self.WordRate`
Because the `if` statements throw an error, you don't need `elif` or `else`, you can just assume that if there was no error it will move forward.